network/transport/grpc/connection.go

```diff
@@ -90,6 +90,12 @@ type Connection interface {
 
 	// IsProtocolConnected returns whether the given protocol is active on the connection.
 	IsProtocolConnected(protocol Protocol) bool
+
+	// ErrorStatus returns the status when the connection closed with an error or nil otherwise
+	ErrorStatus() *status.Status
+
+	// SetErrorStatus sets the status when the connection closed with an error
+	SetErrorStatus(status *status.Status)
```

Comment:
I don't think the interface should allow access to this func. Replace it with `atomic.Pointer` and it can go anyways